Documentation/Evolution/UnicodeForStringProcessing.md

```diff
@@ -0,0 +1,677 @@
+# Unicode for String Processing
+
+Proposal: [SE-NNNN](NNNN-filename.md)  
+Authors: [Nate Cook](https://github.com/natecook1000), [Michael Ilseman](https://github.com/milseman)  
+Review Manager: TBD  
+Implementation: [apple/swift-experimental-string-processing][repo]  
+Status: **Draft**
+
+
+## Introduction
+
+This proposal describes `Regex`'s rich Unicode support during regular expression matching, along with the character classes and options and that define that behavior.
+
+## Motivation
+
+Character classes in regular expressions include metacharacters like `\d` to match a digit, `\s` to match whitespace, and `.` to match any character. Individual literal characters can also be thought of as character classes, as they at least match themselves, and, in case-insensitive matching, their case-toggled counterpart. For the purpose of this work, then, we consider a *character class* to be any part of a regular expression literal that can match an actual component of a string.
+
+Operating over classes of characters is a vital component of string processing. Swift's `String` provides, by default, a view of `Character`s or [extended grapheme clusters][graphemes] whose comparison honors [Unicode canonical equivalence][canoneq].
+
+```swift
+let str = "Cafe\u{301}" // "Café"
+str == "Café"           // true
+str.dropLast()          // "Caf"
+str.last == "é"         // true (precomposed e with acute accent)
+str.last == "e\u{301}"  // true (e followed by composing acute accent)
+```
+
+At a regular expression's simplest, without metacharacters or special features, matching should behave like a test for equality. A string should always match a regular expression that simply contains the same characters.
```

Comment:
Throughout this section, you can drop "should". It's not that it should behave this way, it _does_ behave this way.